server/src/UnifiedDataStore.cc

```diff
@@ -379,14 +379,9 @@ void UnifiedDataStore::reset(void)
 
 UnifiedDataStore *UnifiedDataStore::getInstance(void)
 {
-	if (Impl::instance)
-		return Impl::instance;
-
-	Impl::mutex.lock();
+	AutoMutex lock(&Impl::getInstanceMutex);
```

Comment:
mlpl::AutoMutex is enough here.
But,  I think that we should use `std::lock_guard` instead.